pkgs/tools/graphics/vulkan-cts/default.nix

```diff
@@ -0,0 +1,143 @@
+, libxcb
+, makeWrapper
+, ninja
+, pkg-config
+, python3
+, vulkan-loader
+, wayland
+, wayland-protocols
+, zlib
+}:
+let
+  renderdoc = fetchurl {
+    url = "https://raw.githubusercontent.com/baldurk/renderdoc/v1.1/renderdoc/api/app/renderdoc_app.h";
+    hash = "sha256-57XwqlsbDq3GOhxiTAyn9a8TOqhX1qQnGw7z0L22ho4=";
+  };
+
+  amber = fetchFromGitHub {
+    owner = "google";
+    repo = "amber";
+    rev = "8b145a6c89dcdb4ec28173339dd176fb7b6f43ed";
+    hash = "sha256-+xFYlUs13khT6r475eJJ+XS875h2sb+YbJ8ZN4MOSAA=";
+  };
+  jsoncpp = fetchFromGitHub {
+    owner = "open-source-parsers";
+    repo = "jsoncpp";
+    rev = "9059f5cad030ba11d37818847443a53918c327b1";
+    hash = "sha256-m0tz8w8HbtDitx3Qkn3Rxj/XhASiJVkThdeBxIwv3WI=";
+  };
+  glslang = fetchFromGitHub {
+    owner = "KhronosGroup";
+    repo = "glslang";
+    rev = "22d39cd684d136a81778cc17a0226ffad40d1cee";
+    hash = "sha256-6LplxN7HOMK1NfeD32P5JAMpCBlouttxLEOT/XTVpLw=";
+  };
+  spirv-tools = fetchFromGitHub {
+    owner = "KhronosGroup";
+    repo = "SPIRV-Tools";
+    rev = "b930e734ea198b7aabbbf04ee1562cf6f57962f0";
+    hash = "sha256-NWpFSRoxtYWi+hLUt9gpw0YScM3shcUwv9yUmbivRb0=";
+  };
+  spirv-headers = fetchFromGitHub {
+    owner = "KhronosGroup";
+    repo = "SPIRV-Headers";
+    rev = "36c0c1596225e728bd49abb7ef56a3953e7ed468";
+    hash = "sha256-t1UMJnYONWOtOxc9zUgxr901QFNvqkgurjpFA8UzhYc=";
+  };
+  vulkan-docs = fetchFromGitHub {
+    owner = "KhronosGroup";
+    repo = "Vulkan-Docs";
+    rev = "135da3a538263ef0d194cab25e2bb091119bdc42";
+    hash = "sha256-VZ8JxIuOEG7IjsVcsJOcC+EQeZbd16/+czLcO9t7dY4=";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vulkan-cts";
+  version = "1.3.4.1";
+
+  src = fetchFromGitHub {
+    owner = "KhronosGroup";
+    repo = "VK-GL-CTS";
+    rev = "${finalAttrs.pname}-${finalAttrs.version}";
+    hash = "sha256-XUFlYdudyRqa6iupB8N5QkUpumasyLLQEWcr4M4uP1g=";
+  };
+
+  outputs = [ "out" "lib" ];
+
+  prePatch = ''
+    mkdir -p external/renderdoc/src external/spirv-headers external/vulkan-docs
+
+    cp -r ${renderdoc} external/renderdoc/src/renderdoc_app.h
+
+    cp -r ${amber} external/amber/src
+    cp -r ${jsoncpp} external/jsoncpp/src
+    cp -r ${glslang} external/glslang/src
+    cp -r ${spirv-tools} external/spirv-tools/src
+    cp -r ${spirv-headers} external/spirv-headers/src
+    cp -r ${vulkan-docs} external/vulkan-docs/src
+    chmod u+w -R external
+  '';
+
+  buildInputs = [
+    libdrm
+    libffi
+    libglvnd
+    libpng
+    libX11
+    libXau
+    libXdmcp
+    libxcb
+    spirv-headers
+    spirv-tools
+    wayland
+    wayland-protocols
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    glslang
+    makeWrapper
+    ninja
+    pkg-config
+    python3
+  ];
+
+  # Fix cts cmake not coping with absolute install dirs
+  cmakeFlags = [
+    "-DCMAKE_INSTALL_BINDIR=bin"
+    "-DCMAKE_INSTALL_LIBDIR=lib"
+    "-DCMAKE_INSTALL_INCLUDEDIR=include"
+  ];
+
+  postInstall = ''
+    mv $out $lib
```

Comment:
Because CMake installs all the libs by default.
So we get
```
$ ls result-lib/
dr-xr-xr-x     - root root  1 Jan  1970  result-lib
dr-xr-xr-x     - root root  1 Jan  1970 └──  lib
.r--r--r--  24Ki root root  1 Jan  1970    ├──  libGenericCodeGen.a
.r--r--r--  39Ki root root  1 Jan  1970    ├──  libglslang.a
.r--r--r--   852 root root  1 Jan  1970    ├──  libHLSL.a
.r--r--r-- 480Ki root root  1 Jan  1970    ├──  libjsoncpp.a
.r--r--r-- 5.6Mi root root  1 Jan  1970    ├──  libMachineIndependent.a
.r--r--r-- 3.5Ki root root  1 Jan  1970    ├──  libOGLCompiler.a
.r--r--r-- 5.0Ki root root  1 Jan  1970    ├──  libOSDependent.a
.r--r--r-- 236Ki root root  1 Jan  1970    ├──  libSPIRV-Tools-diff.a
.r--r--r-- 163Ki root root  1 Jan  1970    ├──  libSPIRV-Tools-link.a
.r--r--r-- 219Ki root root  1 Jan  1970    ├──  libSPIRV-Tools-lint.a
.r--r--r-- 9.5Mi root root  1 Jan  1970    ├──  libSPIRV-Tools-opt.a
.r--r--r-- 852Ki root root  1 Jan  1970    ├──  libSPIRV-Tools-reduce.a
.r-xr-xr-x 1.6Mi root root  1 Jan  1970    ├──  libSPIRV-Tools-shared.so
.r--r--r-- 2.6Mi root root  1 Jan  1970    ├──  libSPIRV-Tools.a
.r--r--r-- 1.1Mi root root  1 Jan  1970    ├──  libSPIRV.a
.r--r--r-- 390Ki root root  1 Jan  1970    └──  libSPVRemapper.a
```
and

```
$ ls result/
dr-xr-xr-x       - root root  1 Jan  1970  result
dr-xr-xr-x       - root root  1 Jan  1970 ├──  archive-dir
dr-xr-xr-x       - root root  1 Jan  1970 │  └──  vulkan
…
dr-xr-xr-x       - root root  1 Jan  1970 ├──  bin
.r-xr-xr-x     380 root root  1 Jan  1970 │  ├──  deqp-vk
.r-xr-xr-x    50Mi root root  1 Jan  1970 │  └──  deqp-vksc
…
```

I can remove the -lib output and replace the mv if `rm -rf $out` if you prefer that.